src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro128StarStarImpl.cs

```diff
@@ -90,46 +95,42 @@ public override int Next()
 
             public override int Next(int maxValue)
             {
-                if (maxValue > 1)
+                ulong randomProduct = (ulong)maxValue * NextUInt32();
+                uint lowPart = (uint)(randomProduct & uint.MaxValue);
```

Comment:
For those following along, this is implementing `m mod 2**32` with `m & (2**32-1)`

Which is more efficient than `m % 2**32` currently: https://github.com/dotnet/runtime/issues/11442#issuecomment-1363190619

Reply:
might be worth a comment.